src/components/component.ts

```diff
@@ -356,15 +360,17 @@ export class Component {
 
   private _generateClipPath() {
     // The clip path will prevent content from overflowing its Component space.
-    // HACKHACK: IE <=9 does not respect the HTML base element in SVG.
-    // They don't need the current URL in the clip path reference.
-    let prefix = /MSIE [5-9]/.test(navigator.userAgent) ? "" : document.location.href;
-    prefix = prefix.split("#")[0]; // To fix cases where an anchor tag was used
     let clipPathId = Utils.DOM.generateUniqueClipPathId();
-    this._element.attr("clip-path", "url(\"" + prefix + "#" + clipPathId + "\")");
-    let clipPathParent = this._boxContainer.append("clipPath")
-                                           .attr("id", clipPathId);
+    let clipPathParent = this._boxContainer.append("clipPath").attr("id", clipPathId);
     this._addBox("clip-rect", clipPathParent);
+    this._updateClipPath = () => {
+      // HACKHACK: IE <= 9 does not respect the HTML base element in SVG.
+      // They don't need the current URL in the clip path reference.
+      let prefix = /MSIE [5-9]/.test(navigator.userAgent) ? "" : document.location.href;
+      prefix = prefix.split("#")[0]; // To fix cases where an anchor tag was used
+      this._element.attr("clip-path", "url(\"" + prefix + "#" + clipPathId + "\")");
+    };
```

Comment:
Note that the pain point that I am concerned about is mostly from the developer end (where developers may need to hold additional cost to trade for the fact that we get nice security around a variable that should essentially never change) and nothing from the user end (these endpoints are not exposed).